test/driver-test/driver-test.cpp

```diff
@@ -0,0 +1,312 @@
+/*
+* driver-test.cpp
+* Nick DeMarco
+* (c) 2020 sigcpp https://sigcpp.github.io. See LICENSE.MD
+*
+* Attribution and copyright notice must be retained.
+* - Attribution may be augmented to include additional authors
+* - Copyright notice cannot be altered
+* Attribution and copyright info may be relocated but they must be conspicuous.
+*
+* Tester for test-stl-lite driver
+*/
+
+#include "../verifiers.h"
+#include "../options.h"
+#include "../tester.h"
+#include "../utils.h"
+
+#include <iostream>
+#include <string_view>
+#include <vector>
+
+void driver_test();
+void test_get_options();
+void reset_default_options(Options& o);
+void compare(const Options& received, const Options& expected, const std::string& test_case);
+std::vector<std::pair<std::string, Options>> generate_cmdline_option_pairs();
+
+void driver_test()
+{
+	test_get_options();
+}
+
+void test_get_options()
+{
+	char* args[100];
+
+	const std::vector<std::pair<std::string, Options>> args_and_options = generate_cmdline_option_pairs();
+	std::size_t cmd_line_number{ 0 };
+	for (auto iter = args_and_options.begin(); iter != args_and_options.end(); ++iter)
```

Comment:
Curious about the choice to use iterator. Why not use range-for?

Reply:
You are right, a range-for would be much better here.

Reply:
can just use `auto` instead of `std::vector<std::pair<std::string, Options>>` in L43